DesignSafe-CI/portal
# Task/des 2679 toolbar trash and rename buttons

## Overview: ##
Add Trash Button - user must be logged in (authenticated), more than one file can be moved to the trash. Confirmation pop-ups appear in the bottom left of the screen. 
Add Rename Button - user must be logged in (authenticated), only one file can be renamed at a time. When the rename button is clicked, a Rename modal pops up. The three things I'm validating for are: not-blank, does not allow certain characters (& or @ for example), not the same as the original name. Confirmation pop-ups appear in the bottom left of the screen. 

## PR Status: ##


## Related Jira tickets: ##

* [DES-2679](https://tacc-main.atlassian.net/browse/DES-2679)
* [DES-2680](https://tacc-main.atlassian.net/browse/DES-2680)

## Summary of Changes: ##
Created a useTrash.ts and useRename.ts
Created a RenameModal.tsx
Updated the DatafilesToolbar.tsx

## Testing Steps: ##
1. Log in using steps for "[React Setup for V3 Development.](https://tacc-main.atlassian.net/wiki/spaces/UP/pages/6659524/Local+Development+Environment+Setup)" The instructions are at the bottom of the wiki page.
2. Trash: Select 1, 2, 3+ files to move to the Trash. Verify they move. Verify the pop-up notification shows for each file. 
3. Rename: Verify you can only select one file at a time to rename. Try breaking the 3 validation rules: blank input, bad characters ($,%,&,etc), and same name as original. When you rename a file, verify that the new name appears and the pop-up show for the file.

## UI Photos:
<img width="828" alt="Screen Shot 2024-02-19 at 2 56 57 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/8e87114c-93cc-4c40-80e0-908c511b0986">
<img width="836" alt="Screen Shot 2024-02-19 at 2 57 18 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/fd9047c9-9f88-494b-979e-395a269e8041">


## Notes: ##
